src/main/java/health/ere/ps/service/cetp/CETPServerHandler.java

```diff
@@ -47,35 +50,59 @@ public void handlerRemoved(ChannelHandlerContext ctx) {
-                    : ((de.gematik.ws.conn.eventservice.wsdl.v7.FaultMessage) e).getFaultInfo().getTrace().get(0).getCode().toString();
-                cardlinkWebsocketClient.sendJson("vsdmSensorData", Map.of("slotId", slotId, "ctId", ctId, "endTime", endTime, "err", code));
+            Optional<String> cardHandleOpt = event.getMessage().getParameter().stream()
+                .filter(p -> p.getKey().equals("CardHandle"))
+                .map(Event.Message.Parameter::getValue)
+                .findFirst();
+
+            Optional<String> slotIdOpt = event.getMessage().getParameter().stream()
+                .filter(p -> p.getKey().equals("SlotID"))
+                .map(Event.Message.Parameter::getValue)
+                .findFirst();
+
+            Optional<String> ctIdOpt = event.getMessage().getParameter().stream()
+                .filter(p -> p.getKey().equals("CtID"))
+                .map(Event.Message.Parameter::getValue)
+                .findFirst();
+
+            if (cardHandleOpt.isPresent() && slotIdOpt.isPresent() && ctIdOpt.isPresent()) {
+                String cardHandle = cardHandleOpt.get();
+                Integer slotId = Integer.parseInt(slotIdOpt.get());
+                String ctId = ctIdOpt.get();
+                Long endTime = System.currentTimeMillis();
+                Pair<Bundle, String> pair;
+                try {
+                    pair = pharmacyService.getEPrescriptionsForCardHandle(cardHandle, null, null);
+                    Bundle bundle = pair.getKey();
+                    String eventId = pair.getValue();
+                    String xml = parser.encodeToString(bundle);
+                    cardlinkWebsocketClient.sendJson("eRezeptTokensFromAVS", Map.of("slotId", slotId, "ctId", ctId, "tokens", xml));
+
+                    JsonArrayBuilder bundles = prepareBundles(bundle);
+                    cardlinkWebsocketClient.sendJson("eRezeptBundlesFromAVS", Map.of("slotId", slotId, "ctId", ctId, "bundles", bundles));
+
+                    cardlinkWebsocketClient.sendJson("vsdmSensorData", Map.of("slotId", slotId, "ctId", ctId, "endTime", endTime, "eventId", eventId));
+
+                } catch (FaultMessage | de.gematik.ws.conn.eventservice.wsdl.v7.FaultMessage e) {
+                    log.log(Level.WARNING, "Could not get prescription for Bundle", e);
+                    String code = e instanceof FaultMessage
+                        ? ((FaultMessage) e).getFaultInfo().getTrace().get(0).getCode().toString()
+                        : ((de.gematik.ws.conn.eventservice.wsdl.v7.FaultMessage) e).getFaultInfo().getTrace().get(0).getCode().toString();
+                    cardlinkWebsocketClient.sendJson("vsdmSensorData", Map.of("slotId", slotId, "ctId", ctId, "endTime", endTime, "err", code));
+
+                    Pair<String, String> details = getThrowableDetails(e);
+                    cardlinkWebsocketClient.sendJson(
```

Comment:
Please use the same structure like genericError message with slotId and ctId added:

```
[{"type":"genericErrorMessage","ctId":"[ctId]","slotId":"[slotId]","payload":"eyJlcnJvciI6IkFsbCBzbG90cyBhcmUgZnVsbCBmb3IgZGV2aWNlIHdpdGggaWQ6IDE1Iiwic3RhY2t0cmFjZSI6bnVsbH0="},null,null]
```

Reply:
payload is encrypted and contains ctId, slotId, error, stacktrace

Reply:
for avs erezept error we need use case different from `genericErrorMessage` to avoid clash with existing cardlink logic